opennebula/resource_opennebula_service_template.go

```diff
@@ -0,0 +1,395 @@
+package opennebula
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+
+	"github.com/OpenNebula/one/src/oca/go/src/goca"
+	srv_tmpl "github.com/OpenNebula/one/src/oca/go/src/goca/schemas/service_template"
+)
+
+func resourceOpennebulaServiceTemplate() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceOpennebulaServiceTemplateCreate,
+		Read:   resourceOpennebulaServiceTemplateRead,
+		Exists: resourceOpennebulaServiceTemplateExists,
+		Update: resourceOpennebulaServiceTemplateUpdate,
+		Delete: resourceOpennebulaServiceTemplateDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "Name of the Service Template",
+			},
+			"template": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Service Template body in json format",
+			},
+			"permissions": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "Permissions for the Service Template (in Unix format, owner-group-other, use-manage-admin)",
+				ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
+					value := v.(string)
+
+					if len(value) != 3 {
+						errors = append(errors, fmt.Errorf("%q has specify 3 permission sets: owner-group-other", k))
+					}
+
+					all := true
+					for _, c := range strings.Split(value, "") {
+						if c < "0" || c > "7" {
+							all = false
+						}
+					}
+					if !all {
+						errors = append(errors, fmt.Errorf("Each character in %q should specify a Unix-like permission set with a number from 0 to 7", k))
+					}
+
+					return
+				},
+			},
+			"uid": {
+				Type:        schema.TypeInt,
+				Optional:    true,
```

Comment:
remove this line.
Adding "Optional" means the user can set this while this value is always computed by ONE

Reply:
What I've done is reuse those paremeters as input to, so in case they are set a `chown` operation is performed during the creation so once the resource is available it will have the specified owner. It works the same for groups.

Reply:
I missed to remove this comment sorry

Reply:
np!